File: app/src/main/java/com/like/livedatabus/sample/SecondActivity.java
package com.like.livedatabus.sample;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.like.livedatabus.LiveDataBus;
import com.like.livedatabus.sample.databinding.ActivitySecondBinding;
import com.like.livedatabus_annotations.BusObserver;

public class SecondActivity extends AppCompatActivity {
    private ActivitySecondBinding mBinding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_second);
        LiveDataBus.register(this);
        Log.e("LiveDataBus", "SecondActivity onCreate");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        LiveDataBus.unregister(this);
        Log.e("LiveDataBus", "SecondActivity onDestroy");
    }

    public void changeData1(View view) {
        LiveDataBus.post("like1", 100);
    }

    public void changeData2(View view) {
        LiveDataBus.post("like2", new User("name", 18));
    }

    @BusObserver(value = "like1", isSticky = true)
    public void observer1(int i) {
        Log.e("LiveDataBus", "SecondActivity onChanged tag=like1");
        mBinding.tv1.setText(String.valueOf(i));
    }

    @BusObserver(value = "like2", requestCode = "re")
    public void observer2(User u) {
        Log.e("LiveDataBus", "SecondActivity onChanged tag=like2");
        mBinding.tv2.setText(u.toString());
    }

}

File: app/src/main/java/com/like/floweventbus/sample/SecondActivity.kt
package com.like.floweventbus.sample

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.like.floweventbus.FlowEventBus
import com.like.floweventbus.TAG
import com.like.floweventbus.sample.databinding.ActivitySecondBinding
import com.like.floweventbus_annotations.BusObserver

class SecondActivity : AppCompatActivity() {
    private lateinit var mBinding: ActivitySecondBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_second)
        FlowEventBus.register(this)
        Log.e(TAG, "SecondActivity onCreate")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.e(TAG, "SecondActivity onDestroy")
    }

    fun changeData1(view: View?) {
        FlowEventBus.post("like4", 100)
        FlowEventBus.post("like2", "re", User("name", 18))
    }

    fun unregister(view: View?) {
        FlowEventBus.unregister(this)
    }

    @BusObserver(value = ["like4"], isSticky = true)
    fun observer1(data: Int) {
        Log.e(TAG, "SecondActivity observer1 tag=like1 isSticky=true 数据：$data")
        mBinding.tv1.text = data.toString()
    }

    @BusObserver(value = ["like2"], requestCode = "re")
    fun observer2(data: User) {
        Log.e(TAG, "SecondActivity observer2 tag=like2 requestCode=re 数据：$data")
        mBinding.tv2.text = data.toString()
    }

}
